File: 2021/11/03/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumNumbers(root *TreeNode) int {
	s := 0

	var dfs func(*TreeNode, int)
	dfs = func(root *TreeNode, t int) {
		if root == nil {
			return
		}

		t = t*10 + root.Val

		if root.Left == nil && root.Right == nil {
			s += t
			return
		}

		dfs(root.Left, t)
		dfs(root.Right, t)
	}

	dfs(root, 0)

	return s
}

File: 2021/11/03/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.s = 0
        def dfs(root, t):
            if root == None:
                return 0
            
            t = t*10 + root.val
            if root.left == None and root.right == None:
                self.s += t
                return
            
            dfs(root.left, t)
            dfs(root.right, t)
        dfs(root, 0)
        return self.s

            